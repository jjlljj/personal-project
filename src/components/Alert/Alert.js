import React from 'react';
import { createAlert } from 'react-redux-alerts';
 
const Alert = ({ message, close }) => (
    <div className='Alert'>
        {message}
        <button onClick={close}> Click to Dismiss me </button>
    </div>
);
 
export default createAlert({
    alertName: 'Alert'
})(Alert);