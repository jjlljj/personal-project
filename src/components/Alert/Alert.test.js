/* eslint-disable */
import React from 'react'
import { shallow } from 'enzyme'
import { Alert } from './Alert'

describe('Alert', () => {
  let renderedComponent
  let mockMessage
  let mockClose

  beforeEach(() => {
    mockMessage = "I am the message"
    mockClose = jest.fn()
    renderedComponent = shallow(<Alert message={mockMessage} close={mockClose} />)
  })

  it('should match snapshot', () => {

    expect(renderedComponent).toMatchSnapshot()
  })
})
