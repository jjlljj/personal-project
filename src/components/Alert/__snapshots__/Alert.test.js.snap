// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    close={[Function]}
    message="I am the message"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "I am the message",
      "className": "alert",
    },
    "ref": null,
    "rendered": "I am the message",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am the message",
        "className": "alert",
      },
      "ref": null,
      "rendered": "I am the message",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
